File: Secao07 - Comandos de repetição/Exercicios em Python/Exercicio04.py
# -*- coding: utf-8 -*-

maior = -99999
menor = 99999
soma = 0

for num in range(1, 11):
    
    num = int(input("Informe o {0}º valor: ".format(num)))
    if num > 0 :
        if num > maior:
            maior = num
        if num < menor:
            menor = num
        soma = soma + num
    else:
        print("O valor informado é negativo. Por favor, informe um valor positivo.\n")
    
media = soma / 10
print(" ")    
print("O maior valor é {0}".format(maior))
print("O menor valor é {0}".format(menor))
print("A media entre os valores é {0}".format(media))

File: Secao07 - Comandos de repetição/Exercicios em Java/Exercicio04.java
package br.com.udemy.secao07;

import java.util.Scanner;

public class Exercicio04 {

	public static void main(String[] args) {
		Scanner teclado = new Scanner(System.in);
		int soma = 0, maior = -99999, menor = 99999, num;
		float media;
		
		for(int i = 1; i <=10 ; i++) {
			System.out.print("informe o " + i + "º valor: ");
			num = teclado.nextInt();
			
			if(num > 0) {
				if(num > maior) {
					maior = num;
				}
				if(num < menor) {
					menor = num;
				}
				soma = soma + num;
			}else{
				i--;
			}
			}
		
		media = soma / 10;
		
		System.out.println(" ");
		System.out.println("O maior valor é: " + maior);
		System.out.println("O menor valor é: " + menor);
		System.out.println("A media entre todos os numeros é: " + media);
		
		teclado.close();

	}

}
